openvalidators/reward/blacklist.py

```diff
@@ -29,9 +29,16 @@ class Blacklist( BaseRewardModel ):
     def name(self) -> str: return "blacklist"
 
     def reward( self, prompt: str, completion: str, name: str ) -> float:
-        if completion in blacklist: return 0.0
-        else: return 1
+        if completion in blacklist: 
+            return 0.0
+        
+        if completion == prompt:
```

Comment:
I'm not keen on this form of blacklist. It's very rigid and brittle, and is just doing the job of a relevance+reward model. A few improvements would be

- Regex matching
- [Fuzzy](https://pypi.org/project/fuzzywuzzy/) matching

If we are going to continue as-is, we can at least improve the coverage by replacing strictly equals with substring matching.
```python
if any(b in completion for b in blacklist):
    ...
```